File: parth/Week1/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::vector<int> result;
        std::unordered_map<int, int> umap;
        for(int i=0; i<nums.size(); i++){
            if(umap.contains(target - nums[i])){
                result.push_back(i);
                result.push_back(umap[target-nums[i]]);
                break;
            }
            umap.insert({nums[i], i});
        }
        return result;
    }
};

File: Manali/Week1/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = []
        dictionary = {}
        for i in range(len(nums)):
            if target-nums[i] in dictionary:
                l.append(i)
                l.append(dictionary[target-nums[i]])
                return l
            else:
                dictionary[nums[i]] = i
        return l